src/main/java/com/denizenscript/denizencore/utilities/debugging/DebugInternals.java

```diff
@@ -424,61 +417,35 @@ public static void internalFinalOutputPath(String message, Debuggable caller, bo
     /** Used for "log" to get class names. */
     public static final Map<Class<?>, String> classNameCache = new WeakHashMap<>();
 
-    /** Used for "log" to get class names. This class janks access to SecurityManager to be open to the DebugInternals class.
-     * Note: the format of this code is extremely unstable and jank as it relates to Java security internals stuff, do not touch unless you 100% know what you're doing. */
-    public static class SecurityManagerTrick extends SecurityManager {
-        @Override
-        @SuppressWarnings("rawtypes")
-        protected Class[] getClassContext() {
-            return super.getClassContext();
-        }
-    }
-
-    /** Helper to get the current class context. */
-    public static Class[] getClassContext() {
-        return new SecurityManagerTrick().getClassContext();
-    }
-
     /** Used for "log" to get class names. */
     public static boolean canGetClass = true;
 
     /** Helper to get a class name with less JVM overhead using a cache. */
     public static String getClassNameOpti(Class<?> clazz) {
-        String className = classNameCache.get(clazz);
-        if (className == null) {
-            classNameCache.put(clazz, className = clazz.getSimpleName());
-        }
-        return className;
+        return classNameCache.computeIfAbsent(clazz, k -> clazz.getSimpleName());
     }
 
-    /** Helper to get the calling class for a 'log' message. */
-    public static Class<?> getCallerClass() {
+    /** Helper to get the caller for a 'log' message. */
+    public static StackWalker.StackFrame getCallerStackFrame() {
         if (!canGetClass) {
             return null;
         }
         try {
-            if (canGetClass) {
-                Class[] classes = new SecurityManagerTrick().getClassContext();
-                for (int i = 2; i < classes.length; i++) {
-                    if (classes[i] != DebugInternals.class && classes[i] != Debug.class) {
-                        return classes[i];
-                    }
-                }
-            }
+            return StackWalker.getInstance(StackWalker.Option.RETAIN_CLASS_REFERENCE).walk(stackFrameStream -> stackFrameStream.skip(3).findFirst().orElse(null));

```

Comment:
This change removed protections against call format changes in favor of hardcoding a constant that assumes it will never be used differently from how it currently is.
Please reinstate the loop.

Reply:
It should now be identical to the existing behavior